### PR TITLE
Add categories and clips

### PR DESCRIPTION
**Pull request by michaelarnauts**
This PR implements:
* Categories under the channel menu. They can link to an episode or to a program.
* Clips under the programs.

### CODE REVIEW

File: resources/lib/modules/channels.py [hunk 1]
```diff
@@ -108,3 +125,60 @@ def show_channel_menu(key):
...
+        listing_episodes = []
+        for item in category.episodes:
+            # We don't have the Program Name without making a request to the page, so we use CACHE_AUTO instead of CACHE_ONLY.
+            # This will make a request for each item in this view (about 12 items), but it goes quite fast.
+            # Results are cached, so this will only happen once.
+            _LOGGER.error(item)
+            episode = self._api.get_episode(channel, item.path, CACHE_AUTO)
+
+            if episode:
+                # episode.cover = item.cover  # We have a better cover
```
Comment (michaelarnauts) [inline resources/lib/modules/channels.py:179]: i need to remove this line

**Comment by michaelarnauts** [comment]
I'll go ahead and merge this already.

**Comment by dagwieers** [comment]
I think this change broke all our Kodi favourites and watched/resumepoints :-(

If the routing needs to be changed, please make sure that this does not break in the future (as far as this can be predicted ;-)). Otherwise a new major release needs to be made for breaking changes.

We have about 30 or 40 favourites saved for future "consumption"...

**Comment by michaelarnauts** [comment]
Sorry, that's right. There were inconsistencies with the paths that I've fixed. I should probably have mentioned that in the PR.

This is indeed something that should not happen often.

You could fix this faster for now by editting the bookmarks.xml file somewhere (not even sure about the name) with a regex search and replace. That saves you the trouble of adding all bookmarks again.

**Pull request merged**